docs/Deployment.md

```diff
@@ -0,0 +1,663 @@
+/root/sequence_mapping
+/election
+```
+
+Monitoring
+----------
+
+The CT Log server includes support for two monitoring systems:
+
+ - [Google Cloud Monitoring](https://cloud.google.com/monitoring/) provides a
+   push-based system (where the CT Log regularly pushes information/statistics
+   to the monitoring system).
+ - [Prometheus](https://prometheus.io/) (the default) is a pull-based system:
+   the Prometheus system regularly polls the monitored system for
+   information/statistics.
+
+### GCM
+
+Support for Google Cloud Monitoring (GCM) is enabled by setting the
+`--monitoring=gcm` flag on the command line; when this is enabled,
+`ct-server` will push metrics to GCM.
+
+TODO: describe how to configure this
+
+
+### Prometheus
+
+Prometheus support is enabled by setting the `--monitoring=prometheus` flag on
+the command line; when this is enabled, `ct-server` will export metrics at
+`${CT_LOG_URL}/metrics` and Prometheus should be configured to scrape this
+page (and similar pages exported by `etcd`).
+
+Prometheus configuration is covered in the
+[Prometheus documentation](https://prometheus.io/docs/operating/configuration/).
+
+
+Front-End Setup
+---------------
+
+As the web-server that is built into the `ct-server` binary only has limited
+functionality, it is normal to set up the overall CT Log system with a set of
+front-end web servers that act as
+[reverse proxies](https://en.wikipedia.org/wiki/Reverse_proxy) and load
+balancers.  In particular, the CT Log binary does not support HTTPS so the
+front-end server acts as a
+[TLS termination proxy](https://en.wikipedia.org/wiki/TLS_termination_proxy).
+
+Setup and configuration of these reverse-proxy instances is beyond the scope of
+this document, but note that cloud environments often provide this functionality
+(e.g. [Google Cloud Platform](https://cloud.google.com/compute/docs/load-balancing/http/),
+[Amazon EC2](http://aws.amazon.com/documentation/elastic-load-balancing/)).
+
+
+Standalone Setup
+----------------
+
+It can be helpful to run a CT Log instance locally, for experimentation and to
+troubleshoot any problems with the binaries and configuration.  This section
+collates the steps involved.
+
+First, create a private key `privkey.pem` (with corresponding public key
+`pubkey.pem`) and CA certificates file `ca-roots.pem` as
+[described](#private-key-generation) [above](#ca-certificates).
+
+Next, set up a single, local `etcd` instance and populate as [above](#etcd-setup):
+
+```console
+% etcd > etcd.out 2>&1 &  # default port 2379
+% cpp/tools/prepare_etcd.sh localhost 2379 privkey.pem
+... # and wait...
+% etcdctl get /root/serving_sth  # check STH is populated
+CAASIgogjdFMhilPo5O5inGTLeQV8sOSOVYmjKIPnwJD2Mhh0dkYs/Kikc4qIAAqIOOwxEKY/BwUmvv0yJlvuSQnrkHkZJuTTKSVmRt4UrhVMk0IBBADGkcwRQIhAITYZ8KTbGktnvxA+i44w5SkoiUSGQB0u/e2reQsZG7YAiBl9la11f2ifs6h2/mmJ6JLLJZlhCM5E1311YOgZjo8Xg=
+```
+
+The default configuration for the set of CT Log instances requires a minimum
```

Comment:
It's fairly easy to simply change the cluster configuration to only require 1 log instance too, something like:
```bash
echo "minimum_serving_nodes: 1" > /tmp/cluster_config && ../cpp/tools/ct-clustertool set_config --etcd_servers=localhost:4001 --cluster_config=/tmp/cluster_config
```

one wrinkle is that if you try this immediately after the `prepare_etcd.sh` step then you may have to wait for a minute or so while the `ct_clustertool` regains mastership.